definitions/3.0.0/multiFormatSchema.json

```diff
@@ -54,7 +54,7 @@
         "description": "Definition of the message payload. It can be of any type but defaults to Schema Object. It MUST match the schema format defined in schemaFormat, including the encoding type. E.g., Avro should be inlined as either a YAML or JSON object instead of as a string to be parsed as YAML or JSON. Non-JSON-based schemas (e.g., Protobuf or XSD) MUST be inlined as a string."
       }
     },
-    "allOf": [
+    "anyOf": [
```

Comment:
After a deep investigation, I concluded this **cannot** be changed to `anyOf` and **should** stay as `allOf`.

The reason is that `anyOf` will try to match with any schema in the array. The problem is that since those schemas are `if`, and an `if` conditional fails, it is considered an empty schema, which is equivalent to `true` assertion of that element in the `anyOf` array.

See this example I made by extracting the logic in place and reducing the complexity for test purposes: https://www.jsonschemavalidator.net/s/q0RfvVqb

As you can see, the `if` is failing (doesn't match the schemaFormat), but the `anyOf` is resulting in a true assertion. Totally against what we expect.

#### How can we fix then the issue?

Let's read what the JSON Schema validator says:

![Slack_TGf24bNa@2x](https://github.com/asyncapi/spec-json-schemas/assets/1083296/cde988c6-5d52-4a42-afa1-0e096ccaaff4)

The issue comes not at this line, but at [this other one](https://github.com/asyncapi/spec-json-schemas/blob/master/schemas/3.0.0-without-%24id.json#L2820-L2827).

```json
"oneOf": [
    {
        "$ref": "#/definitions/Reference"
    },
    {
        "$ref": "#/definitions/schema"
    }
]
```

The point is that the keyword `$ref` is both defined in the `Reference` schema and in the `schema`, which is a superset of JSON Schema and it does include the `$ref` keyword itself.
That's why JSON Schema validation says the schema is matching with more than one schema in this`oneOf`.

By changing that `oneOf` to `anyOf`, we ensure that by matching with any of those schemas will be considered a valid doc.

I spent some time trying to find edge-cases for the change I'm suggesting. For example, what if we change how our Reference Object (`$ref`) look like? Then, it will differ from JSON Schema provided one, and we could set it back then to `oneOf` if needed.

Btw, this change should be propagated to the `oneOf` in OpenAPI since OpenAPI schema is also a superset of JSON Schema and it includes the `$ref` property.


Additionally, I investigated a bit if we could get rid of those `if/then` and I believe we could just use `allOf` with all the possible cases as schemas right directly, without the `if/then`, including the case when the `schemaFormat` is just an unknown `string`.
But this, is out of the scope of this issue I believe. Just for the record.